modules/solid_mechanics/test/tests/dynamics/time_integration/direct_central_difference_multiVarBC.i

```diff
@@ -0,0 +1,128 @@
+###########################################################
+# This is a simple test with a time-dependent problem
+# demonstrating the use of a central difference with a
+# direct calculation of acceleration.
+#
+# Testing that the first and second time derivatives
+# are calculated correctly using the Central Difference Direct
+# method
+###########################################################
+
+[GlobalParams]
+    displacements = 'disp_x disp_y'
+[]
+
+[Mesh]
+    type = GeneratedMesh
+    dim = 2
+    xmin = -1
+    xmax = 1
+    ymin = -1
+    ymax = 1
+    nx = 4
+    ny = 4
+[]
+
+[Variables]
+    [disp_x]
+    []
+    [disp_y]
+    []
+[]
+
+[Functions]
+    [forcing_fn]
+        type = ParsedFunction
+        expression = 't'
+    []
+[]
+
+[Kernels]
+    [DynamicSolidMechanics]
+        displacements = 'disp_x disp_y'
+    []
+    [massmatrix]
+        type = MassMatrix
+        density = density
+        matrix_tags = 'system'
+        variable = disp_x
+    []
+    [massmatrix_y]
```

Comment:
Doesn't this second mass matrix effectively double the density?

Reply:
It shouldn't
They are each associated with a separate variable. So one populates the mass on the dofs for dispx and one for dispy.

Reply:
I see. This is interesting though, and probably warrants a future PR: If the displacement variables all have the same family and order, their mass matrices should be identical. There's gotta be a way to avoid assembling the matrices 3 times in 3D (twice in 2D).

Could you create an issue for this? I'll then mark this as resolved.

Reply:
Fair point, it'll be the same on all three dofs. Certainly a performance bottleneck. I'll put up an issue.

Reply:
#28436